libr/util/str.c

```diff
@@ -2928,3 +2929,21 @@ R_API char* r_str_highlight(char *str, const char *word, const char *color) {
 	}
 	return strdup (o);
 }
+
+R_API int r_str_scpy (char *dst, const char *src, int count) {
+  if (count <= 0) {
+    return -E2BIG;
+  }
+  int ret = 0;
+  while (count) {
+    if (!(dst[ret] = src[ret])) {
+      return ret;
+    }
+    ret++;
+    count--;
+  }
+  if (ret) {
+    dst[ret-1] = '\0';
+  }
+  return -E2BIG;
```

Comment:
`r_str_ncpy` requires `r_str_ncpy (dst, src, sizeof dst - 1);` the minus one makes it a little awkward to use.